gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,261 @@
+    // collect marker row for the given run-id
+    // Example Query:
+    // SELECT gcRunId FROM nessie.db1.`identified_results@someGcRef`
+    //    WHERE gcRunId = '1a088e65-a6ea-42e9-819d-025f4e22eaec' AND type = 'GC_MARK'
+    Dataset<Row> markerRow =
+        sql(
+            "SELECT %s FROM %s WHERE %s = '%s' AND %s = '%s'",
+            COL_GC_RUN_ID,
+            //
+            catalogAndTableWithRefName,
+            //
+            COL_GC_RUN_ID,
+            runId,
+            //
+            COL_TYPE,
+            TYPE_GC_MARKER);
+    if (markerRow.isEmpty()) {
+      return markerRow;
+    }
+    // Example Query:
+    // SELECT * FROM nessie.db1.`identified_results@someGcRef`
+    //    WHERE gcRunId = '9e809dc9-ac35-41c9-b1c0-5ebcf452ea2d' AND type = 'GC_CONTENT'
+    return sql(
+        "SELECT * FROM %s WHERE %s = '%s' AND %s = '%s'",
+        catalogAndTableWithRefName,
+        //
+        COL_GC_RUN_ID,
+        runId,
+        //
+        COL_TYPE,
+        TYPE_CONTENT);
+  }
+
+  /**
+   * Get the last written run id which is having a marker row.
+   *
+   * @return latest completed run id.
+   */
+  public String getLatestCompletedRunID() {
+    // collect marker row for the last written run id
+    // Example query:
+    // SELECT gcRunId FROM nessie.db1.`identified_results@someGcRef
+    //    WHERE type = 'GC_MARK' ORDER BY gcRunStart DESC LIMIT 1
+    List<Row> markerRows =
+        sql(
+                "SELECT %s FROM %s WHERE %s = '%s' ORDER BY %s DESC LIMIT 1",
+                COL_GC_RUN_ID,
+                //
+                catalogAndTableWithRefName,
+                //
+                COL_TYPE,
+                TYPE_GC_MARKER,
+                //
+                COL_GC_RUN_START)
+            .collectAsList();
+    return markerRows.isEmpty() ? null : markerRows.get(0).getString(0);
+  }
+
+  /**
+   * Convert each task result into a spark row and store it in the output table. Write a marker row
+   * in the end to indicate that the task is completed.
+   *
+   * @param rowDataset content rows to be written.
+   * @param runId run id of current identify task.
+   * @param startedAt task start time.
+   */
+  void writeToOutputTable(Dataset<Row> rowDataset, String runId, Timestamp startedAt) {
+    try {
+      // write content rows
+      rowDataset.writeTo(catalogAndTableWithRefName).append();
+      // write marker row
+      sparkSession
+          .createDataFrame(Collections.singletonList(createMarkerRow(runId, startedAt)), schema)
+          .writeTo(catalogAndTableWithRefName)
+          .append();
+    } catch (NoSuchTableException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  static Row getContentRowVariablePart(
+      String contentId, String contentType, long snapshotId, String refName, String hash) {
+    return RowFactory.create(
+        IdentifiedResultsRepo.TYPE_CONTENT,
+        // the fixed value columns (runId and startTime)
```

Comment:
Why not just pass those values downstream instead this apporach?

Reply:
Really better create the `Row` as it will look like instead of manipulating it later.

Reply:
No problem, I will change it.

Reply:
fixed.